python/cudf/cudf/_lib/csv.pyx

```diff
@@ -403,17 +403,23 @@ cpdef write_csv(
     bool header=True,
     str line_terminator="\n",
     int rows_per_chunk=8,
+    bool index=True,
 ):
     """
     Cython function to call into libcudf API, see `write_csv`.
 
     See Also
     --------
-    cudf.io.csv.write_csv
+    cudf.io.csv.to_csv
     """
+    all_cols = table._columns
+    all_names = table._column_names
 
-    # Index already been reset and added as main column, so just data_view
-    cdef table_view input_table_view = table.data_view()
+    if index is True:
+        all_cols = table._index._data.columns + all_cols
+        all_names = table._index.names + all_names
+
+    cdef table_view input_table_view = make_table_view(all_cols)
```

Comment:
I want to say we have a method you can call against `table` that returns a `table_view` with all columns instead of extracting the columns to a list and creating the `table_view` from that.

Reply:
Yes, fixed this.